app/views/advanced_messenger/update_message_read_by_users.js.erb

```diff
@@ -0,0 +1,6 @@
+var node = $("#<%=@message.id == @message.root.id ? 'topic-wrapper'.html_safe : ('message-' + @message.id.to_s).html_safe%>");  
+renderExpandAndReadIndicatorForCurrentUser("<%= @message.id %>", "update_message_read_by_users", node, <%= @message.read_by_users.html_safe %>, "<%= User.current.id %>", "<%= escape_javascript(@message.content.to_s[0, 130]) %>");
+renderReadIndicatorsForAllUsers(<%=@message.id%>, node, "<%= User.current.id %>", '<%=l(:label_time_ago, :time => time_tag(@message.created_on).html_safe).html_safe%>', ...<%= raw(getUsersReadStatus(@message.read_by_users).to_json)%>);
+
+renderBottomRightUnreadNotificationsDialog(<%=@unread_notification_statuses[0]%>, <%=@unread_notification_statuses[1]%>, <%=@unread_notification_statuses[2].to_json.html_safe%>, "<%='?r=' + @unread_notification_statuses[0].to_s + '#message-' + @unread_notification_statuses[0].to_s%>");
+renderGlobalUnreadNotificationsIndicator(<%= getUnreadNotificationsForCurrentUserCount()%>);
```

Comment:
This is called only once per page, right? Not for every message/note?

Reply:
Yes. It is called once per page. It is called again only if some read status changes (e.g. the user reads a message)